src/resources/postcss/components/_view-selector.pcss

```diff
@@ -0,0 +1,273 @@
+		text-align: left;
+		width: 100%;
+
+		@media (--viewport-medium) {
+			padding: var(--spacer-3) var(--spacer-4);
+		}
+
+		&:hover,
+		&:focus {
+			border-bottom-color: var(--color-accent-primary);
+
+			&:before {
+				opacity: 1;
+			}
+		}
+
+		&:before {
+			background-color: var(--color-accent-primary);
+			content: '';
+			display: block;
+			height: 2px;
+			opacity: 0;
+			position: absolute;
+			top: 100%;
+			transform: translateY(var(--spacer-0));
+			width: calc(100% - (2 * var(--spacer-0)));
+
+			@media (--viewport-medium) {
+				content: none;
+			}
+		}
+
+		&:after {
+			@media (--viewport-medium) {
+				background-image: svg-inline(caret-down);
+				background-repeat: no-repeat;
+				background-size: contain;
+				content: '';
+				display: block;
+				flex: none;
+				height: 6px;
+				margin-left: var(--spacer-1);
+				width: 10px;
+			}
+		}
+	}
+
+	.tribe-events-c-view-selector__button--active {
+
+		&:after {
+			@media (--viewport-medium) {
+				background-image: svg-inline(caret-up);
+			}
+		}
+	}
+
+	.tribe-events-c-view-selector__button-icon {
+		content: '';
+		display: block;
+		height: 21px;
+		width: 21px;
+
+		@media (--viewport-medium) {
+			@mixin hidden;
+		}
+	}
+
+	.tribe-events-c-view-selector__button-text {
+		@mixin visually-hide;
+
+		@media (--viewport-medium) {
+			@mixin visually-show;
+			@mixin desktop-body-2-bold;
+
+			flex: auto;
+		}
+	}
+
+	.tribe-events-c-view-selector__content {
+		display: none;
+		background-color: var(--color-background);
+		border: 1px solid var(--color-border-secondary);
+		border-radius: 4px;
+		box-shadow: var(--box-shadow-tooltip);
+		padding: calc(var(--spacer-4) / 2) var(--spacer-1);
+		position: absolute;
+		right: 0;
+		top: 100%;
+		transform: translateY(var(--spacer-2));
+		width: 168px;
+
+		@media (--viewport-medium) {
+			box-shadow: var(--box-shadow-default);
+			transform: translateY(var(--spacer-1));
+			width: 100%;
+		}
+	}
+
+	/* .tribe-events-c-view-selector__list {} */
+
+	/* .tribe-events-c-view-selector__list-item {} */
+
+	.tribe-events-c-view-selector__list-item-link {
+		align-items: center;
+		border-radius: 4px;
+		display: flex;
+		padding: var(--spacer-1) calc(var(--spacer-4) / 2);
+
+		@media (--viewport-medium) {
+			padding: var(--spacer-0) var(--spacer-2);
+		}
+
+		&:hover,
+		&:focus {
+
+			.tribe-events-c-view-selector__list-item-text {
+				color: var(--color-accent-secondary-hover);
+				font-weight: var(--font-weight-bold);
+			}
+		}
+	}
+
+	.tribe-events-c-view-selector__list-item-link--active {
+		background-color: var(--color-background-secondary);
+
+		.tribe-events-c-view-selector__list-item-text {
+			font-weight: var(--font-weight-bold);
+		}
+	}
+
+	.tribe-events-c-view-selector__list-item-icon {
+		flex: none;
+		height: 18px;
+		margin-right: 12px;
+		width: 18px;
+
+		@media (--viewport-medium) {
+			@mixin hidden;
+		}
+	}
+
+	.tribe-events-c-view-selector__list-item-text {
+		@mixin desktop-body-2;
```

Comment:
I feel like in other places we're using the common class, instead of the mixin on the newly defined class. Why do we have it like that here, instead of including the b2 class from common in the markup?

Reply:
The `tribe-common-b2` class has a mobile font size and a desktop font size. The designs here call for the same font size on both mobile and desktop.